buildtools/dep/dep.go

```diff
@@ -22,11 +43,32 @@ type Project struct {
 	Version  string
 }
 
-// A Lockfile contains the contents of a dep lockfile. Lockfiles are resolvers.
-type Lockfile struct {
-	Projects []Project
+//Resolve determines if the Gopkg.toml dep file has any rules that apply
+func (r Resolver) Resolve(importpath string) (pkg.Import, error) {
+	if r.isIgnored(importpath) {
+		return pkg.Import{}, buildtools.ErrNoRevisionForPackageOk
+	}
 
-	normalized map[string]pkg.Import // A normalized map of package import paths to revisions.
+	return r.Lockfile.Resolve(importpath)
+}
+
+//isIgnored finds packages to ignore while handling the possibility of a wildcard
+func (r Resolver) isIgnored(importpath string) bool {
+
+	for _, matcher := range r.TOMLFile.Ignored {
+		if matcher == importpath {
+			return true
```

Comment:
[nit] Technically, you should do this _after_ the wildcard check. This might break on packages with stars in the name. Are those package names valid?

Reply:
They are invalid on Github. Theoretically you could create a local package with a wildcard in the name but that should break dep before it got to us. Also, the * is only allowed at the end of the package by dep, so if there was a wildcard at any point in the name it would be ignored by dep which allows us to ignore the behavior as well.